presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpillerFactory.java

```diff
@@ -80,6 +90,29 @@ private static ListeningExecutorService createExecutorServiceOfSize(int nThreads
         return MoreExecutors.listeningDecorator(executorService);
     }
 
+    @PostConstruct
+    public void cleanupOldSpillFiles()
+    {
+        spillPaths.forEach(this::cleanupOldSpillFiles);
+    }
+
+    private void cleanupOldSpillFiles(Path path)
+    {
+        Arrays.stream(path.toFile().listFiles(new PatternFilenameFilter(SPILL_FILE_PATTERN))).forEach(
+                spillFile -> {
+                    try {
+                        log.info("Deleting old spill file: " + spillFile);
+                        if (!spillFile.delete()) {
+                            throw new RuntimeException("cannot delete");
```

Comment:
Are you sure you want to fail starting the server when old files can't be deleted? I would log a warning and continue.